oauth2client/contrib/gce.py

```diff
@@ -85,32 +121,110 @@ class AppAssertionCredentials(AssertionCredentials):
     information to generate and refresh its own access tokens.
     """
 
-    @util.positional(2)
-    def __init__(self, scope='', **kwargs):
+    @util.positional(3)
+    def __init__(self,
+                 scope=None,
+                 service_account_email='default',
+                 service_account_info=None,
+                 **kwargs):
         """Constructor for AppAssertionCredentials
 
         Args:
-            scope: string or iterable of strings, scope(s) of the credentials
-                   being requested. Using this argument will have no effect on
-                   the actual scopes for tokens requested. These scopes are
-                   set at VM instance creation time and won't change.
+            scope:
+                string or iterable of strings, scope(s) of the credentials
+                being requested. Using this argument will have no effect on
+                the actual scopes for tokens requested. These scopes are
+                set at VM instance creation time and won't change.
+            service_account_email:
+                the email for these credentials. This can be used with custom
+                service accounts, or left blank to use the default service
+                account for the instance. Usually the compute engine service
+                account.
+            service_account_info:
+                Deserialized JSON object, returned by self.service_account_info
```

Comment:
Sorry for my vague comment. I'd say just update to this value to something more descriptive. How about:

```
            service_account_info:
                (Optional) Dictionary containing information about a
                service account (typically deserialized from JSON).
                If passed in, will be used as ``service_account_info``
                property. Otherwise, the Compute Engine metadata server
                will be used to determine this value.
```